server/middlewares/validations/user.js

```diff
@@ -0,0 +1,52 @@
+import Joi from "joi";
+
+const createUserSchema = Joi.object().keys({
+  firstname: Joi.string()
+    .min(2)
+    .required(),
+  lastname: Joi.string()
+    .min(2)
+    .required(),
+  email: Joi.string()
+    .min(2)
+    .required(),
+  password: Joi.string()
+    .regex(/^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-.]).{6,}$/)
+    .min(7)
+    .error(errors => {
+      return {
+        message:
+          "password must be atleast 6 chars with atleast 1 uppercase, 1 number, & 1 special char"
+      };
+    })
+    .required()
+});
+
+/**
+ * Validate User Input on Registration
+ * @param {object} req The request body object
+ * @param {object} res
+ * @param {function} next
+ * @returns {object} Response object
+ */
+const validateUserInput = async (req, res, next) => {
```

Comment:
I feel this logic can be re-used by other validation schemas, please refactor this;

Reply:
I will like to get more clarification on this, please.

Reply:
Certainly, the way this has been defined you have tightly coupled it with the createUserSchema. So now If I am writing another schema e.g. createArticle I would need a separate function for that as well. I need the logic to be reusable so any Joischema can use it.

Reply:
Okay. It means that the method name and the file name will also need to change to signify that it is a generic method.